client/components/ShareButton.js

```diff
@@ -21,33 +22,70 @@ const useStyles = ((canShare) => makeStyles((theme) => ({
 export default () => {
   const [state, setState] = useState({showSnackbar: false});
   const shareData = {url: window.location.href};
-  const classes = useStyles(navigator.canShare && navigator.canShare(shareData))();
+  const canShare = navigator.canShare && navigator.canShare(shareData);
+  const classes = useStyles(
+    canShare || (navigator.clipboard && navigator.clipboard.writeText))();
```

Comment:
nit: extract the clipboard stuff into a new variable?
```js
const canShare = navigator.canShare && navigator.canShare(shareData);
const canCopy = navigator.clipboard && navigator.clipboard.writeText;

const classes = useStyles(canShare || canCopy)();
```

Reply:
Done.